src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -718,12 +719,6 @@ private SingleTypeDeclaration VisitTypeDeclaration(TypeDeclarationSyntax node, D
                 }
             }
 
-            if (node is ExtensionDeclarationSyntax extension &&
-                extension.ImplicitOrExplicitKeyword.IsKind(SyntaxKind.ExplicitKeyword))
-            {
-                declFlags |= SingleTypeDeclaration.TypeDeclarationFlags.IsExplicitExtension;
```

Comment:
>declFlags |= SingleTypeDeclaration.TypeDeclarationFlags.IsExplicitExtension; [](http://example.com/codeflow?start=16&length=76)

I am not sure if the optimization is worth it, given that we are passing `declFlags` to `GetNonTypeMemberNames` which technically might look at the values. Therefore, the code becomes somewhat fragile long term. It looks, however, that `GetNonTypeMemberNames` doesn't actually check value of incoming `declFlags` today. Therefore, if we want to keep the optimization, we could change that parameter to an `out` parameter to enforce the fact that no one looks at the decl flags until they are fully populated.

Reply:
done.